src/library/scala/collection/mutable/HashSet.scala

```diff
@@ -28,64 +29,227 @@ package mutable
   * @define mayNotTerminateInf
   * @define willNotTerminateInf
   */
-final class HashSet[A]
+final class HashSet[A](initialCapacity: Int, loadFactor: Double)
   extends AbstractSet[A]
     with SetOps[A, HashSet, HashSet[A]]
     with StrictOptimizedIterableOps[A, HashSet, HashSet[A]] {
 
-  @transient private[this] var table = new FlatHashTable[A]
+  def this() = this(HashSet.defaultInitialCapacity, HashSet.defaultLoadFactor)
 
-  // Used by scala-java8-compat (private[mutable] erases to public, so Java code can access it)
-  private[mutable] def getTable: FlatHashTable[A] = table
-  override def iterator: Iterator[A] = table.iterator
+  import HashSet.Node
 
-  override def iterableFactory: IterableFactory[HashSet] = HashSet
+  /** The actual hash table. */
+  private[this] var table = new Array[Node[A]](tableSizeFor(initialCapacity))
```

Comment:
Did you consider an `Array[Any]` to avoid the `Node` allocation if there's no collision?

Reply:
Ah, then the hash code would not be cached. That would cause more work on resizing, I guess?

Reply:
That's the hybrid approach I tried first. It performed well. I abandoned it in order to cache the hashcode because that's what we do in all the other standard hashtable-based collections (in particular, `mutable.HashMap` and `immutable.HashSet` + `HashMap`)

Reply:
OK, thanks! I agree consistency is important here.